tornado/ioloop.py

```diff
@@ -1162,24 +1201,30 @@ class _Timeout(object):
     """An IOLoop timeout, a UNIX timestamp and a callback"""
 
     # Reduce memory overhead when there are lots of pending callbacks
-    __slots__ = ['deadline', 'callback', 'tdeadline']
+    __slots__ = ['deadline', 'callback', 'seqno']
 
     def __init__(self, deadline, callback, io_loop):
-        if not isinstance(deadline, numbers.Real):
+        if not isinstance(deadline, Real):
             raise TypeError("Unsupported deadline %r" % deadline)
         self.deadline = deadline
+        self.seqno = next(io_loop._timeout_counter)
         self.callback = callback
-        self.tdeadline = (deadline, next(io_loop._timeout_counter))
 
     # Comparison methods to sort by deadline, with object id as a tiebreaker
     # to guarantee a consistent ordering.  The heapq module uses __le__
     # in python2.5, and __lt__ in 2.6+ (sort() and most other comparisons
     # use __lt__).
+    @cython.locals(self = '_Timeout', other = '_Timeout')
     def __lt__(self, other):
-        return self.tdeadline < other.tdeadline
+        return (self.deadline < other.deadline
```

Comment:
Shouldn't you cast the values to `Timeout`?

Reply:
They're declared as such in the method annotation